File: app/src/classes/Course.js
/**
 * Course Class
 * Demonstrates: Encapsulation
 * Represents a course in the system
 */

export class Course {
  // Private attributes - Encapsulation
  #courseCode;
  #courseName;

  constructor(courseCode, courseName) {
    this.#courseCode = courseCode;
    this.#courseName = courseName;
  }

  // Getters - Encapsulation
  getCourseCode() {
    return this.#courseCode;
  }

  getCourseName() {
    return this.#courseName;
  }

  // Setters - Encapsulation
  setCourseCode(courseCode) {
    this.#courseCode = courseCode;
  }

  setCourseName(courseName) {
    this.#courseName = courseName;
  }

  // Display course information
  display() {
    return {
      code: this.#courseCode,
      name: this.#courseName
    };
  }

  // Get formatted display string
  toString() {
    return `${this.#courseName} (${this.#courseCode})`;
  }
}

File: java-app/src/Course.java
/**
 * Course Class
 * Demonstrates: Encapsulation
 * Represents a course in the system
 */
public class Course {
    // Encapsulation: Private attributes
    private String courseCode;
    private String courseName;

    // Constructor
    public Course(String courseCode, String courseName) {
        this.courseCode = courseCode;
        this.courseName = courseName;
    }

    // Getters - Encapsulation
    public String getCourseCode() {
        return courseCode;
    }

    public String getCourseName() {
        return courseName;
    }

    // Setters - Encapsulation
    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    public void setCourseName(String courseName) {
        this.courseName = courseName;
    }

    // Display course information
    public void display() {
        System.out.println("Course Code: " + courseCode + " | Course Name: " + courseName);
    }
}
